### PR TITLE
SSCS-10798 Reworking Grant Postponement Hearing Process

### PR DESCRIPTION
**Pull request by TeaWithLucas**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-10798

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsWindowMapping.java [hunk 1]
```diff
@@ -69,8 +71,9 @@ public static LocalDateTime getFirstDateTimeMustBe() {
         return null;
     }
 
-    public static boolean isPostponementGranted(SscsCaseData caseData) {
-        String requestSelected = caseData.getPostponementRequest().getActionPostponementRequestSelected();
-        return GRANT.getValue().equals(requestSelected);
+    public static boolean isCasePostponed(SscsCaseData caseData) {
+        PostponementRequest postponementRequest = caseData.getPostponementRequest();
+        return !isYes(postponementRequest.getUnprocessedPostponementRequest())
```
[the remark this answers is not in the document]
Comment (TeaWithLucas) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsWindowMapping.java]: Yes, I have changed it to isNoOrNull

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsEventMappers.java [hunk 2]
```diff
@@ -1,20 +1,50 @@
...
+        } else if (shouldCaseBeRelisted(caseData)) {
+            return EventType.READY_TO_LIST;
+        } else if (shouldCaseBeNotListable(caseData)) {
+            return EventType.UPDATE_NOT_LISTABLE;
+        } else {
+            return null;
+        }
+    }
+
+    public static boolean shouldCaseBeRelisted(SscsCaseData caseData) {
```
[the remark this answers is not in the document]
Comment (dattipoeHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsEventMappers.java:36]: Are these methods not being used in here alone? We need to move away from adding more static classes

**Review by dattipoeHMCTS — DISMISSED** [review]

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/service/HearingsServiceHelperTest.java [hunk 3]
```diff
@@ -246,4 +250,39 @@ void testIsCaseHearingRequestedOrAwaitingListingNonMatching(HmcStatus value) {
...
+        HearingsServiceHelper.clearTransientFields(wrapper);
+
+        PostponementRequest result = wrapper.getCaseData().getPostponementRequest();
+
+        assertThat(result)
+            .hasAllNullFieldsOrPropertiesExcept("unprocessedPostponementRequest");
+        assertThat(result.getUnprocessedPostponementRequest()).isEqualTo(NO);
+    }
+
+    @DisplayName("If case is not postponed clearTransientFields doesnt not clear PostponementRequest field")
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/helper/service/HearingsServiceHelperTest.java:271]: Double negative in the DisplayName - should be "does not"?

**Comment by amoslv1** [comment]
One small naming error in a test

**Review by amoslv1 — DISMISSED** [review]
One small naming error in a test

**Pull request closed without merging**